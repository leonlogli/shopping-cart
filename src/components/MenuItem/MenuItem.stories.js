import React from 'react'
import Mail from '@material-ui/icons/Mail'

import MenuItem from '.'

export default {
  title: 'MenuItem',
}

export const Default = () => <MenuItem text="Option 1" />

export const withIcon = () => <MenuItem text="Email" icon={<Mail />} />

export const asLink = () => <MenuItem href="/test" text="Inbox" />
