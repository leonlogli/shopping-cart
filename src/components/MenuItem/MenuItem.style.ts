import styled from 'styled-components'
import MenuItem from '@material-ui/core/MenuItem'

const MenuItemRoot = styled(MenuItem)`
  .MuiSvgIcon-root {
    font-size: 1.25rem;
  }
`

export { MenuItemRoot }
