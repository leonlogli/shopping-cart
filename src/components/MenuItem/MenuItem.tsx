import React from 'react'
import { MenuItemProps as MuiMenuItemProps } from '@material-ui/core/MenuItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'

import { BoxProps } from '../Layout'
import { Link, LinkProps } from '../Link'

import { MenuItemRoot } from './MenuItem.style'
import { Text } from '../Typography'

export type MenuItemProps = Omit<MuiMenuItemProps, 'button'> &
  Pick<BoxProps, 'component'> &
  LinkProps & {
    icon?: React.ReactElement
    text: string
  }

/**
 * Mobile side nav menu item
 */
const MenuItem = (props: MenuItemProps) => {
  const { href, as, to, text, icon, component, onClick, ...other } = props

  const renderLink = React.useMemo(
    () =>
      React.forwardRef<any, Omit<LinkProps, 'href'>>((itemProps, ref) => (
        <Link
          href={href}
          as={as}
          ref={ref as any}
          underline="none"
          color="inherit"
          to={to}
          {...itemProps}
        />
      )),
    [href]
  )

  const componentProp: any = { component: href ? renderLink : component }

  return (
    <MenuItemRoot
      button={(!!href || !!onClick) as any}
      component={(href ? renderLink : component) as any}
      onClick={onClick as any}
      {...componentProp}
      {...other}
    >
      {icon && <ListItemIcon>{icon}</ListItemIcon>}
      <Text variant="inherit">{text}</Text>
    </MenuItemRoot>
  )
}

export { MenuItem }
export default MenuItem
