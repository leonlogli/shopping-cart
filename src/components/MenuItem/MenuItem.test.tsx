import React from 'react'
import Mail from '@material-ui/icons/Mail'
import { mount } from 'enzyme'

import MenuItem from '.'

describe('<MenuItem />', () => {
  const wrapper = mount(<MenuItem text="text" />)

  it('should render children', () => {
    expect(wrapper.text()).toEqual('text')
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have icon', () => {
    wrapper.setProps({ icon: <Mail /> } as any)

    expect(wrapper.find('svg').length).toEqual(1)
  })

  it('should be a link', () => {
    wrapper.setProps({ href: '/' } as any)

    expect(wrapper.find('a').length).toEqual(1)
  })
})
