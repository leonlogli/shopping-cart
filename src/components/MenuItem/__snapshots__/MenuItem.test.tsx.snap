// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> should match a snapshot 1`] = `
<MenuItem
  text="text"
>
  <Styled(WithStyles(ForwardRef(MenuItem)))
    button={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      button={false}
      className="KpA-dS"
    >
      <ForwardRef(MenuItem)
        button={false}
        className="KpA-dS"
        classes={
          Object {
            "dense": "MuiMenuItem-dense",
            "gutters": "MuiMenuItem-gutters",
            "root": "MuiMenuItem-root",
            "selected": "Mui-selected",
          }
        }
      >
        <WithStyles(ForwardRef(ListItem))
          button={false}
          className="MuiMenuItem-root KpA-dS MuiMenuItem-gutters"
          classes={
            Object {
              "dense": "MuiMenuItem-dense",
            }
          }
          component="li"
          disableGutters={false}
          role="menuitem"
          tabIndex={-1}
        >
          <ForwardRef(ListItem)
            button={false}
            className="MuiMenuItem-root KpA-dS MuiMenuItem-gutters"
            classes={
              Object {
                "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
                "button": "MuiListItem-button",
                "container": "MuiListItem-container",
                "dense": "MuiListItem-dense MuiMenuItem-dense",
                "disabled": "Mui-disabled",
                "divider": "MuiListItem-divider",
                "focusVisible": "Mui-focusVisible",
                "gutters": "MuiListItem-gutters",
                "root": "MuiListItem-root",
                "secondaryAction": "MuiListItem-secondaryAction",
                "selected": "Mui-selected",
              }
            }
            component="li"
            disableGutters={false}
            role="menuitem"
            tabIndex={-1}
          >
            <li
              className="MuiListItem-root MuiMenuItem-root KpA-dS MuiMenuItem-gutters MuiListItem-gutters"
              disabled={false}
              role="menuitem"
              tabIndex={-1}
            >
              <Text
                key=".1"
                variant="inherit"
              >
                <Styled(WithStyles(ForwardRef(Typography)))
                  variant="inherit"
                >
                  <WithStyles(ForwardRef(Typography))
                    className=""
                    variant="inherit"
                  >
                    <ForwardRef(Typography)
                      className=""
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter",
                          "alignJustify": "MuiTypography-alignJustify",
                          "alignLeft": "MuiTypography-alignLeft",
                          "alignRight": "MuiTypography-alignRight",
                          "body1": "MuiTypography-body1",
                          "body2": "MuiTypography-body2",
                          "button": "MuiTypography-button",
                          "caption": "MuiTypography-caption",
                          "colorError": "MuiTypography-colorError",
                          "colorInherit": "MuiTypography-colorInherit",
                          "colorPrimary": "MuiTypography-colorPrimary",
                          "colorSecondary": "MuiTypography-colorSecondary",
                          "colorTextPrimary": "MuiTypography-colorTextPrimary",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary",
                          "displayBlock": "MuiTypography-displayBlock",
                          "displayInline": "MuiTypography-displayInline",
                          "gutterBottom": "MuiTypography-gutterBottom",
                          "h1": "MuiTypography-h1",
                          "h2": "MuiTypography-h2",
                          "h3": "MuiTypography-h3",
                          "h4": "MuiTypography-h4",
                          "h5": "MuiTypography-h5",
                          "h6": "MuiTypography-h6",
                          "noWrap": "MuiTypography-noWrap",
                          "overline": "MuiTypography-overline",
                          "paragraph": "MuiTypography-paragraph",
                          "root": "MuiTypography-root",
                          "srOnly": "MuiTypography-srOnly",
                          "subtitle1": "MuiTypography-subtitle1",
                          "subtitle2": "MuiTypography-subtitle2",
                        }
                      }
                      variant="inherit"
                    >
                      <span
                        className="MuiTypography-root "
                      >
                        text
                      </span>
                    </ForwardRef(Typography)>
                  </WithStyles(ForwardRef(Typography))>
                </Styled(WithStyles(ForwardRef(Typography)))>
              </Text>
            </li>
          </ForwardRef(ListItem)>
        </WithStyles(ForwardRef(ListItem))>
      </ForwardRef(MenuItem)>
    </WithStyles(ForwardRef(MenuItem))>
  </Styled(WithStyles(ForwardRef(MenuItem)))>
</MenuItem>
`;
